password_security/models/res_users.py

```diff
@@ -76,19 +55,36 @@ def _check_password_rules(self, password):
         if not password:
             return True
         company_id = self.company_id
-        password_regex = ['^']
-        if company_id.password_lower:
-            password_regex.append('(?=.*?[a-z])')
-        if company_id.password_upper:
-            password_regex.append('(?=.*?[A-Z])')
-        if company_id.password_numeric:
-            password_regex.append(r'(?=.*?\d)')
-        if company_id.password_special:
-            password_regex.append(r'(?=.*?\W)')
-        password_regex.append('.{%d,}$' % company_id.password_length)
-        if not re.search(''.join(password_regex), password):
-            raise PassError(self.password_match_message())
-        return True
+        message = []
+        if len(list(filter(str.islower, password))
```

Comment:
If you remove the `list()` part, you save a bunch of loops.

Reply:
>len(filter(str.islower, password)) 

is not work in python3